### PR TITLE
Added `getExplosionDamages` to index.d.ts

### PR DESCRIPTION
**Pull request by link-discord**

### CODE REVIEW

**Comment by link-discord** [comment]
While I'm at it I also made it so if `rawDamages` is true in the `explosion.js` plugin it also checks if the armour props are present. Since  using the calculateExplosionDamages function without armour and `rawDamages` disabled it would cause an error

File: index.d.ts [hunk 1]
```diff
@@ -257,6 +257,8 @@ export interface Bot extends TypedEmitter<BotEvents> {
 
   clearControlStates: () => void
 
+  getExplosionDamages: (entity: Entity, position: Vec3, raidus: number, rawDamages?: boolean) => number
```
Comment (u9g) [inline index.d.ts:260]:
```suggestion
  getExplosionDamages: (targetEntity: Entity, position: Vec3, radius: number, rawDamages?: boolean) => number
```

File: lib/plugins/explosion.js [hunk 2]
```diff
@@ -77,7 +77,7 @@ function inject (bot) {
     // The following modifiers are constant for the input targetEntity and doesnt depend
     // on the source position, so if the goal is to compare between positions they can be
     // ignored to save computations
-    if (!rawDamages) {
+    if (!rawDamages && targetEntity.attributes['generic.armor']) {
```
Comment (u9g) [inline lib/plugins/explosion.js:80]: instead of returning undefined maybe we throw here? or return a null?

Comment (link-discord) [reply lib/plugins/explosion.js:80]: The main reason I added this change is so that it doesn't throw. I would say we just return null instead

**Comment by u9g** [comment]
can you add to typings that this returns number or null and to docs, and explain why in docs

**Comment by link-discord** [comment]
> can you add to typings that this returns number or null and to docs, and explain why in docs

Ok I will do it now

**Review by u9g — APPROVED** [review]
LGTM

**Pull request merged**